lib/elixir/lib/code/fragment.ex

```diff
@@ -286,72 +312,290 @@ defmodule Code.Fragment do
         if ?@ in extra do
           :none
         else
-          {rest, _} = strip_spaces(rest, 0)
-          {kind, rest, acc}
+          {rest, count} = strip_spaces(rest, count)
+          {kind, rest, acc, count}
         end
 
       _ ->
         :none
     end
   end
 
-  defp nested_alias(rest, acc) do
-    {rest, _} = strip_spaces(rest, 0)
+  defp nested_alias(rest, count, acc) do
+    {rest, count} = strip_spaces(rest, count)
 
-    case identifier_to_cursor_context(rest, true) do
-      {:alias, prev} -> {:alias, prev ++ '.' ++ acc}
-      _ -> :none
+    case identifier_to_cursor_context(rest, count, true) do
+      {{:alias, prev}, count} -> {{:alias, prev ++ '.' ++ acc}, count}
+      _ -> {:none, 0}
     end
   end
 
-  defp dot(rest, acc) do
-    {rest, _} = strip_spaces(rest, 0)
+  defp dot(rest, count, acc) do
+    {rest, count} = strip_spaces(rest, count)
 
-    case identifier_to_cursor_context(rest, true) do
-      {:local_or_var, prev} -> {:dot, {:var, prev}, acc}
-      {:unquoted_atom, _} = prev -> {:dot, prev, acc}
-      {:alias, _} = prev -> {:dot, prev, acc}
-      {:dot, _, _} = prev -> {:dot, prev, acc}
-      {:module_attribute, _} = prev -> {:dot, prev, acc}
-      _ -> :none
+    case identifier_to_cursor_context(rest, count, true) do
+      {{:local_or_var, var}, count} -> {{:dot, {:var, var}, acc}, count}
+      {{:unquoted_atom, _} = prev, count} -> {{:dot, prev, acc}, count}
+      {{:alias, _} = prev, count} -> {{:dot, prev, acc}, count}
+      {{:dot, _, _} = prev, count} -> {{:dot, prev, acc}, count}
+      {{:module_attribute, _} = prev, count} -> {{:dot, prev, acc}, count}
+      {_, _} -> {:none, 0}
     end
   end
 
-  defp operator([h | rest], acc, call_op?) when h in @operators do
-    operator(rest, [h | acc], call_op?)
+  defp operator([h | rest], count, acc, call_op?) when h in @operators do
+    operator(rest, count + 1, [h | acc], call_op?)
   end
 
-  defp operator(rest, acc, call_op?) when acc in @incomplete_operators do
-    {rest, _} = strip_spaces(rest, 0)
+  defp operator(rest, count, acc, call_op?) when acc in ~w(^^ ~~ ~)c do
+    {rest, dot_count} = strip_spaces(rest, count)
 
     cond do
-      call_op? -> :none
-      match?([?. | rest] when rest == [] or hd(rest) != ?., rest) -> dot(tl(rest), acc)
-      true -> {:operator, acc}
+      call_op? ->
+        {:none, 0}
+
+      match?([?. | rest] when rest == [] or hd(rest) != ?., rest) ->
+        dot(tl(rest), dot_count + 1, acc)
+
+      true ->
+        {{:operator, acc}, count}
     end
   end
 
-  defp operator(rest, acc, _call_op?) do
+  defp operator(rest, count, acc, _call_op?) do
     case :elixir_tokenizer.tokenize(acc, 1, 1, []) do
```

Comment:
Note `:elixir_tokenizer.tokenize/4` returns `{:ok, tokens}` on Elixir v1.12. So backporting this code requires wrapping this call (which is a private API anyway).